### PR TITLE
Closes #48

### LINKED ISSUE #48: Create a dataset loader for BioNLP Shared Task 2009
From http://www.geniaproject.org/shared-tasks/bionlp-shared-task-2009

### PR DESCRIPTION
**Pull request by HallerPatrick**
Please name your PR after the issue it closes. You can use the following line: "Closes #ISSUE-NUMBER" where you replace the ISSUE-NUMBER with the one corresponding to your dataset.

If the following information is NOT present in the issue, please populate:

- **Name:** BioNLP Shared Task 2009
- **Description:** http://www.geniaproject.org/shared-tasks/bionlp-shared-task-2009
- **Paper:** [*Dataset Paper*](https://aclanthology.org/W09-1401/)
- **Data:** [*link to the online home of the dataset*](http://www.geniaproject.org/shared-tasks/bionlp-shared-task-2009)

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: biodatasets/bionlp_shared_task_2009/bionlp_shared_task_2009.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+_URL_BASE = "http://www.nactem.ac.uk/GENIA/current/Shared-tasks/BioNLP-ST-2009/"
+_URLS = {
+    _DATASETNAME: {
+        "train": _URL_BASE + "bionlp09_shared_task_training_data_rev2.tar.gz",
+        "test": _URL_BASE + "bionlp09_shared_task_test_data_without_gold_annotation.tar.gz",
+        "dev": _URL_BASE + "bionlp09_shared_task_development_data_rev1.tar.gz",
+    },
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION, Tasks.EVENT_EXTRACTION]
```
Comment (leonweber) [inline biodatasets/bionlp_shared_task_2009/bionlp_shared_task_2009.py:158]: This should also include coreference resolution

**Review by leonweber — CHANGES_REQUESTED** [review]

**Review by leonweber — APPROVED** [review]
Looks good to me. Failure for test_schema is expected because in all BioNLP shared task datasets test set annotations are hidden and thus missing from our version.
